oracle/pkg/updater/updater.go

```diff
@@ -460,18 +472,65 @@ func (u *Updater) getL1Txns(ctx context.Context, blockNum uint64) (map[string]in
 
 	u.metrics.BlockTxnCacheMisses.Inc()
 
-	blk, err := u.l1Client.BlockByNumber(ctx, big.NewInt(0).SetUint64(blockNum))
+	block, err := u.l1Client.BlockByNumber(ctx, big.NewInt(0).SetUint64(blockNum))
 	if err != nil {
 		return nil, fmt.Errorf("failed to get block by number: %w", err)
 	}
 
-	txnsInBlock := make(map[string]int)
-	for posInBlock, tx := range blk.Transactions() {
-		txnsInBlock[strings.TrimPrefix(tx.Hash().Hex(), "0x")] = posInBlock
+	var txnReceipts sync.Map
+	eg, ctx := errgroup.WithContext(ctx)
+
+	txnsArray := make([]common.Hash, len(block.Transactions()))
+	for i, tx := range block.Transactions() {
+		txnsArray[i] = tx.Hash()
+	}
+
+	bucketSize := (len(txnsArray) + 3) / 4 // Calculate the size of each bucket, rounding up
```

Comment:
So is the desire, that we make 4 Max requests, but each request needs to be of a some maximal size up to 50? or otherwise tuneable factor.

Reply:
I'd rather we optimize for these things if we see them as a bottleneck. My intuition tells me this wouldn't be an issue. We do these types of requests every 12 seconds.

Reply:
Are you talking about evmHelper? because that's not an exported structure?

Reply:
Where is the alternative mechanism in the codebase?

Reply:
I wasn't able to understand what you're referring to by the library, but I've updated the code to bucket in the mechanism you describe. Now it will only send the required number of requests.

Reply:
Ah, that makes a lot of sense!